### PR TITLE
fix: add activate logic

### PR DESCRIPTION
**Pull request by xiaolang124**
1. Add activate environment logic and icon.
2. Add view environment icon.
3. Add switch env to provision modal confirm.

image
image

https://microsoft.sharepoint.com/:w:/t/TeamsDevX/EQrXbG9n1-tGjDo17r4wEVoBjpVGG2OCa1hBs8XLTPLfmQ?e=r3bTEL

### CODE REVIEW

File: packages/fx-core/src/core/middleware/envInfoLoader.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ export function EnvInfoLoaderMW(
         targetEnvName = await askTargetEnvironment(ctx, inputs, allowCreateNewEnv, lastUsedEnvName);
       }
       lastUsedEnvName = targetEnvName ?? lastUsedEnvName;
+      ctx.projectSettings.activeEnvironment = lastUsedEnvName;
```
Comment (qinezh) [inline packages/fx-core/src/core/middleware/envInfoLoader.ts:71]: Let's confirm with UX designer that whether we should update the activate environment with the last used environment name.

File: packages/fx-core/src/core/index.ts [hunk 2]
```diff
@@ -855,6 +856,45 @@ export class FxCore implements Core {
...
+  async activateEnv(
+    env: string,
+    inputs: Inputs,
+    ctx?: CoreHookContext
+  ): Promise<Result<Void, FxError>> {
+    if (!isMultiEnvEnabled() || !ctx!.projectSettings) {
+      return ok(Void);
+    }
+
+    const envConfigs = await environmentManager.listEnvConfigs(inputs.projectPath!);
```
[the remark this answers is not in the document]
Comment (xiaolang124) [reply packages/fx-core/src/core/index.ts:929]: Fixed.

File: packages/fx-core/src/core/index.ts [hunk 3]
```diff
@@ -855,6 +856,45 @@ export class FxCore implements Core {
...
+
+    if (!envConfigs.isErr() && envConfigs.value!.indexOf(env!) < 0) {
```
[the remark this answers is not in the document]
Comment (xiaolang124) [reply packages/fx-core/src/core/index.ts]: Fixed.

File: packages/fx-core/src/core/middleware/envInfoLoader.ts (hunk 1 above)
Comment (xiaolang124) [reply packages/fx-core/src/core/middleware/envInfoLoader.ts]: Changed to new modal confirm.

File: packages/vscode-extension/package.json [hunk 4]
```diff
@@ -344,10 +354,22 @@
...
+        "command": "fx-extension.viewEnvironmentWithIcon",
+        "title": "View environment",
+        "icon": "$(file-symlink-directory)",
+        "enablement": "fx-extension.isMultiEnvEnabled"
+      },
       {
         "command": "fx-extension.activateEnvironment",
         "title": "Activate environment",
         "enablement": "fx-extension.isMultiEnvEnabled"
+      },
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/package.json:367]: 1. Why keep both xxx and xxxWithIcon?
2. Hide the command  in command palette?

File: packages/vscode-extension/src/envTree.ts [hunk 5]
```diff
@@ -34,6 +39,7 @@ export async function registerEnvTreeHandler(): Promise<Result<Void, FxError>> {
           contextValue: "environment",
           icon: "symbol-folder",
           isCustom: false,
+          description: item === activeEnv ? "active" : "",
```
Comment (HuihuiWu-Microsoft) [inline packages/vscode-extension/src/envTree.ts:42]: The string seems visible to users, better move to strings.json

**Review by LongOddCode — APPROVED** [review]

File: packages/vscode-extension/package.json (hunk 4 above)
Comment (xiaolang124) [reply packages/vscode-extension/package.json:367]: For now there are both icon and right click to trigger this feature.

File: packages/vscode-extension/src/envTree.ts (hunk 5 above)
Comment (xiaolang124) [reply packages/vscode-extension/src/envTree.ts:42]: Fixed.

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**